### PR TITLE
[2.1] Fix: libcrmcommon: Drop xmlRegisterNodeDefault() and xmlDeregisterNodeDefault() 

### PR DESCRIPTION
**Pull request by nrwahl2**
Within libxml2, these are tagged with the "deprecated" compiler attribute, causing builds on some bleeding-edge distributions (currently at least debian-experimental) to fail.

Instead, allocate and free private data as needed when creating and freeing XML nodes.

Fixes T838

### CODE REVIEW

File: cts/cli/regression.tools.exp [hunk 1]
```diff
@@ -8739,7 +8739,7 @@ cts-cli
 =#=#=#= End test: Get active shadow instance's file name (copied) (XML) - OK (0) =#=#=#=
 * Passed: crm_shadow     - Get active shadow instance's file name (copied) (XML)
 =#=#=#= Begin test: Get active shadow instance's contents (copied) =#=#=#=
-<cib epoch="1" num_updates="173" admin_epoch="1">
+<cib epoch="1" num_updates="173" admin_epoch="1"> update-origin="cluster01" update-client="crmd" update-user="hacluster" have-quorum="1" dc-uuid="2">
```
Comment (kgaillot) [inline cts/cli/regression.tools.exp:8742]: ?

Comment (nrwahl2) [reply cts/cli/regression.tools.exp:8742]: Same thing happened on main branch. As far as I can tell, some of the sanitization was more aggressive than it needed to be. We can cut this out if you want though.

Comment (kgaillot) [reply cts/cli/regression.tools.exp:8742]: I don't see it on main. Just need to figure out why the `>` is in the wrong place and fix it

Comment (nrwahl2) [reply cts/cli/regression.tools.exp:8742]: https://github.com/ClusterLabs/pacemaker/pull/3642/commits/942c760b06f727091ef4596bb5f7c6f64ed1d94a#diff-6e0f7d808fa417f92613a0760eace3fc813b73adc89f4a228e4a2d320cb2c9e4R93

etc.

Comment (nrwahl2) [reply cts/cli/regression.tools.exp:8742]: Previously we were cutting out everything from `cib-last-written` through the end of the attributes. Now we cut out specific attributes (`cib-last-written`, generation attrs, `validate-with`)

Comment (kgaillot) [reply cts/cli/regression.tools.exp:8742]: It looks good on main. The issue here is the `>` placement

Comment (nrwahl2) [reply cts/cli/regression.tools.exp:8742]: Ohhhh

**Review by kgaillot — APPROVED** [review]

**Pull request merged**